src/main/java/core/basesyntax/serviceimpl/ReportServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportService;
+import java.util.Map;
+
+public class ReportServiceImpl implements ReportService {
+    private static final String SEPARATOR = ",";
+    private static final String HEADER = "fruit,quantity";
+
+    @Override
+    public String createReport() {
+        StringBuilder stringBuilder = new StringBuilder(HEADER);
+        for (Map.Entry<String, Integer> entry : Storage.storage.entrySet()) {
+            stringBuilder.append(System.lineSeparator())
+                    .append(entry.getKey())
+                    .append(SEPARATOR)
+                    .append(entry.getValue());
```

Comment:
While the logic for creating the report seems correct, consider encapsulating the logic for formatting a single entry into a separate method. This will make the `createReport` method more readable and maintain the Single Responsibility Principle, one of the SOLID rules.